pkg/sql/pg_catalog.go

```diff
@@ -834,10 +832,243 @@ var (
+			}
+			if r, ok := fkMatchMap[con.FK.Match]; ok {
+				confmatchtype = r
+			}
+			if conkey, err = colIDArrayToDatum(con.FK.OriginColumnIDs); err != nil {
+				return err
+			}
+			if confkey, err = colIDArrayToDatum(con.FK.ReferencedColumnIDs); err != nil {
+				return err
+			}
+			var buf bytes.Buffer
+			if err := showForeignKeyConstraint(&buf, db.Name, table, con.FK, tableLookup); err != nil {
+				return err
+			}
+			condef = tree.NewDString(buf.String())
+
+		case sqlbase.ConstraintTypeUnique:
+			oid = h.UniqueConstraintOid(db, scName, table, con.Index)
+			contype = conTypeUnique
+			conindid = h.IndexOid(table.ID, con.Index.ID)
+			var err error
+			if conkey, err = colIDArrayToDatum(con.Index.ColumnIDs); err != nil {
+				return err
+			}
+			f := tree.NewFmtCtx(tree.FmtSimple)
+			f.WriteString("UNIQUE (")
+			con.Index.ColNamesFormat(f)
+			f.WriteByte(')')
+			condef = tree.NewDString(f.CloseAndGetString())
+
+		case sqlbase.ConstraintTypeCheck:
+			oid = h.CheckConstraintOid(db, scName, table, con.CheckConstraint)
+			contype = conTypeCheck
+			if conkey, err = colIDArrayToDatum(con.CheckConstraint.ColumnIDs); err != nil {
+				return err
+			}
+			consrc = tree.NewDString(fmt.Sprintf("(%s)", con.Details))
+			conbin = consrc
+			condef = tree.NewDString(fmt.Sprintf("CHECK ((%s))", con.Details))
+		}
+
+		if err := addRow(
+			oid,                  // oid
+			dNameOrNull(conName), // conname
+			namespaceOid,         // connamespace
+			contype,              // contype
+			tree.DBoolFalse,      // condeferrable
+			tree.DBoolFalse,      // condeferred
+			tree.MakeDBool(tree.DBool(!con.Unvalidated)), // convalidated
+			tblOid,         // conrelid
+			oidZero,        // contypid
+			conindid,       // conindid
+			confrelid,      // confrelid
+			confupdtype,    // confupdtype
+			confdeltype,    // confdeltype
+			confmatchtype,  // confmatchtype
+			tree.DBoolTrue, // conislocal
+			zeroVal,        // coninhcount
+			tree.DBoolTrue, // connoinherit
+			conkey,         // conkey
+			confkey,        // confkey
+			tree.DNull,     // conpfeqop
+			tree.DNull,     // conppeqop
+			tree.DNull,     // conffeqop
+			tree.DNull,     // conexclop
+			conbin,         // conbin
+			consrc,         // consrc
+			condef,         // condef
+		); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type oneAtATimeSchemaResolver struct {
+	ctx context.Context
+	p   *planner
+}
+
+func (r oneAtATimeSchemaResolver) getDatabaseByID(id sqlbase.ID) (*DatabaseDescriptor, error) {
+	return r.p.Tables().databaseCache.getDatabaseDescByID(r.ctx, r.p.txn, id)
+}
+
+func (r oneAtATimeSchemaResolver) getTableByID(id sqlbase.ID) (*TableDescriptor, error) {
+	table, err := r.p.LookupTableByID(r.ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	return table.Desc.TableDesc(), nil
+}
+
+// makeAllRelationsPGCatalogTable creates a virtual table that searches through
+// all table descriptors in the system. It automatically adds an index to the
```

Comment:
* It automatically adds a virtual index implementation to the table id column.

I think you want to specify because the user has to actually write the index definition in the create statement themselves